text/0000-arbitrary-self-types-v2.md

```diff
@@ -233,19 +233,27 @@ As noted in the rationale section, the currently nightly implementation implemen
 ## No blanket implementation for `Deref`
 [no-blanket-implementation]: #no-blanket-implementation
 
-The other major approach previously discussed is to have a `Receiver` trait, as proposed in this RFC, but without a blanket implementation for `T: Deref`. Possible advantages:
+The other major approach previously discussed is to have a `Receiver` trait, as proposed in this RFC, but without a blanket implementation for `T: Deref`. Blanket implementations are unusual for core Rust traits, but the authors of this RFC believe it's a powerful tool to reduce user confusion. In particular, a different specification of `Target` for these two traits would sometimes lead the compiler to explore radically different chains of types when determining the candidates for dereferencing and method calls. It's easy to imagine cases where this would be confusing for users. Imagine a type `A` dereferences to `B` yet allows method calls only on `C` (or perhaps more confusingly, allows method calls only on `Pin<&mut B>`).
```

Comment:
Personally I'm less worried about different targets; while exotic you would need to have both permission to implement traits for `A` and `C` _must_ be a crate-local type (I assume this remains true for `impl C` and nothing about the RFC changes it). So in the example you hypothesise with `A`, `B`, and `C`, the implementation is really limited to a single crate rather than split across multiple types.

Reply:
Hmmm. I see what you mean. I need to think about whether that reduces my estimation of the maximum confusion that this could cause.

Reply:
I tried to implement this (receiver trait without a blanket implementation) in rustc so that I could have a crack at building a maximally-confusing example Rust project here. However, as noted below, it turned out that removing the blanket implementation was much more disruptive than expected, and convinced me that it's a big compatibility break if we don't have the blanket implementation.